### PR TITLE
Amalgam8 Demo Walkthroughs

### PR DESCRIPTION
**Pull request by rshriram**
Porting documents from the Amalgam8 examples to the docs site. Splitting the demo walkthrough into multiple easy to consume sections.

Got rid of vagrant deployment. 
Pointing users to install K8S via minikube, and docker directly from the docker site.
Also got rid of Marathon deployment (can add it if needed), since we are not frequently testing it.

### CODE REVIEW

File: _docs/demo/setup.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+   where `<your bluemix space>` is the name of your Bluemix space and
+   `<your route ...>` is a unique route name for `registry` and
+   `controller`. For example `my-space-name-a8-registry` and 
+   `my-space-name-a8-controller`. Make a note of the route names you
+   choose for the next step.
+
+
+1. Download the scripts to automate Bluemix deployment from this
+   [here](https://url_to_bluemix_scripts).
```
Comment (theophoric) [inline _docs/demo/setup.md]: Don't forget to provide a real value for this url stub.

File: _docs/demo/setup.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+order: 1
+---
+
+While Amalgam8 supports multi-tenancy, for the sake of simplicity, this
+demo will setup Amalgam8 in single-tenant mode.
+
+## Download
+
+Download and unzip the latest release of Amalgam8 demos from the
+[Amalgam8 Releases](https://www.github.com/amalgam8/amalgam8/releases)
```
Comment (theophoric) [inline _docs/demo/setup.md:15]: A note about this: the release is called "a8demos-v0.3.1.zip" but the folder inside is still called "examples"

File: _docs/demo/setup.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+_IBM Bluemix_
+
+1. Login to Bluemix and initialize the container environment using 
+
+   ```
+   bluemix login
+   bluemix ic init
+   ```
```
Comment (theophoric) [inline _docs/demo/setup.md:75]: Won't `cf login; cf ic init` work as well?  (provided bluemix containers plugin is installed)

File: _docs/demo/setup.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+_Kubernetes_ on localhost or on Google Cloud
+
+```bash
+kubectl create -f examples/k8s-controlplane.yaml
+```
+
+_IBM Bluemix_
+
```
Comment (theophoric) [inline _docs/demo/setup.md:69]: Add pre-reqs for bluemix?  Cf cli?  Bluemix Cli? Bluemix containers plugin?

File: _docs/demo/setup.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+1. Download the scripts to automate Bluemix deployment from this
+   [here](https://url_to_bluemix_scripts).
+   
+1. Customize the _bluemix.cfg_ in the downloaded folder as follows:
+    * BLUEMIX_REGISTRY_NAMESPACE should be your Bluemix registry namespace
+      obtained from the folowing command:
+
+      ```bash
+      bluemix ic namespace-get
```
Comment (theophoric) [inline _docs/demo/setup.md]: should be `bluemix ic namespace get`

Comment (theophoric) [reply _docs/demo/setup.md]: ( or `cf ic namespace get` )

File: _docs/demo/setup.md (hunk 1 above)
Comment (theophoric) [reply _docs/demo/setup.md]: User has already downloaded the demos release, no?  Just tell them to open the bluemix folder?

File: _docs/demo/setup.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+   `<your route ...>` is a unique route name for `registry` and
+   `controller`. For example `my-space-name-a8-registry` and 
+   `my-space-name-a8-controller`. Make a note of the route names you
+   choose for the next step.
+
+
+1. Download the scripts to automate Bluemix deployment from this
+   [here](https://url_to_bluemix_scripts).
+   
+1. Customize the _bluemix.cfg_ in the downloaded folder as follows:
```
Comment (theophoric) [inline _docs/demo/setup.md]: this file is called `.bluemixrc` in latest release.  Did this change in this pr .. ?

File: _docs/demo/setup.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+      ```
+
+    * REGISTRY_HOSTNAME should be the route name assigned to the registry in the previous step
+
+    * CONTROLLER_HOSTNAME should be the route name assigned to the controller in the previous step
+
+1. Deploy the control plane services (registry and controller) on bluemix.
+
+   ```bash
+   examples/a8-bluemix create controlplane
```
Comment (theophoric) [inline _docs/demo/setup.md:106]: Folder in current demos release is called `bluemix`

File: _docs/demo/setup.md (hunk 6 above)
Comment (theophoric) [reply _docs/demo/setup.md]: Also should either include instructions to `cd` into bluemix folder, or reference all file names relative to the `examples` directory

File: _docs/demo/setup.md (hunk 7 above)
Comment (theophoric) [reply _docs/demo/setup.md:106]: script is called `deploy-controlplane.sh`

File: _docs/demo/setup.md (hunk 1 above)
Comment (rshriram) [reply _docs/demo/setup.md]: fixed

File: _docs/demo/setup.md (hunk 2 above)
Comment (rshriram) [reply _docs/demo/setup.md:15]: fixed

File: _docs/demo/setup.md (hunk 3 above)
Comment (rshriram) [reply _docs/demo/setup.md:75]: Commands for cf cli seem to be different from those for bluemix cli. Esp bx ic groups, etc., which are needed for the tools.

File: _docs/demo/setup.md (hunk 4 above)
Comment (rshriram) [reply _docs/demo/setup.md:69]: done

File: _docs/demo/setup.md (hunk 5 above)
Comment (rshriram) [reply _docs/demo/setup.md]: fixed

File: _docs/demo/setup.md (hunk 6 above)
Comment (rshriram) [reply _docs/demo/setup.md]: it changed to bluemix.cfg . Also there is no folder specific per runtime.

File: _docs/demo/setup.md (hunk 7 above)
Comment (rshriram) [reply _docs/demo/setup.md:106]: fixed. Its called a8-bluemix

File: _docs/demo/bookinfo.md [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+```bash
+kubectl create -f examples/k8s-bookinfo.yaml
+```
+
+_IBM Bluemix_
+
+1. Create Bluemix routes (DNS names) for the registry, controller and the bookinfo app's gateway:  
+
+   ```bash
+   cf create-route <your bluemix space> mybluemix.net -n <your bookinfo route>
```
Comment (dougbeauchene) [inline _docs/demo/bookinfo.md:74]: With the effort to provide a unified CLI in Bluemix this  "cf create-route <your bluemix space> mybluemix.net -n <your bookinfo route>"   should be    "bluemix cf create-route <your bluemix space> mybluemix.net -n <your bookinfo route>"

I have verified putting "bluemix" on the front of the command works.

There is another place in this file where you are using the "bluemix" command, so I think it's a good thing to use "bluemix" here as well to show consistency to the reader.

File: _docs/demo/helloworld.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+
+The above command automatically launches two instances of each version of
+`helloworld`.
+
+_IBM Bluemix_
+
+1. Create Bluemix routes (DNS names) for the registry, controller and the helloworld app's gateway:  
+
+   ```bash
+   cf create-route <your bluemix space> mybluemix.net -n <your helloworld route>
```
Comment (dougbeauchene) [inline _docs/demo/helloworld.md:46]: Same comment as the other file.  "bluemix" is the unified CLI for Bluemix.  So change "cf create-route...." to "bluemix cf create-route...."

**Pull request merged**